engine/src/main/java/org/pentaho/di/pan/delegates/PanTransformationDelegate.java

```diff
@@ -0,0 +1,207 @@
+    for ( Map.Entry<String, String> entry : paramMap.entrySet() ) {
+      transMeta.setParameterValue( entry.getKey(), Const.NVL( entry.getValue(), "" ) );
+    }
+    transMeta.activateParameters();
+
+    // Set the log level
+    if ( executionConfiguration.getLogLevel() != null ) {
+      transMeta.setLogLevel( executionConfiguration.getLogLevel() );
+    }
+
+    // Determine execution type and execute accordingly
+    return executeBasedOnConfiguration( transMeta, executionConfiguration, arguments );
+  }
+
+  /**
+   * Execute transformation based on the execution configuration type.
+   */
+  private Result executeBasedOnConfiguration( TransMeta transMeta,
+                                              TransExecutionConfiguration executionConfiguration,
+                                              String[] arguments ) throws KettleException {
+
+    if ( executionConfiguration.isExecutingLocally() ) {
+      return transformationExecutorServiceMap.get( "LOCAL" ).execute( log, transMeta, repository, executionConfiguration, arguments );
+    } else if ( executionConfiguration.isExecutingRemotely() ) {
+      return transformationExecutorServiceMap.get( "REMOTE" ).execute( log, transMeta, repository, executionConfiguration, arguments );
+    } else if ( executionConfiguration.isExecutingClustered() ) {
+      return transformationExecutorServiceMap.get( "CLUSTERED" ).execute( log, transMeta, repository, executionConfiguration, arguments );
+
+    } else {
+      throw new KettleException( BaseMessages.getString( pkg, "PanTransformationDelegate.Error.NoExecutionTypeSpecified" ) );
+    }
+  }
+
+  /**
+   * Create a default execution configuration for command-line execution.
+   */
+  public static TransExecutionConfiguration createDefaultExecutionConfiguration() {
+    TransExecutionConfiguration config = new TransExecutionConfiguration();
+
+    // Set defaults for command-line execution
+    config.setExecutingLocally( true );
+    config.setExecutingRemotely( false );
+    config.setExecutingClustered( false );
+    config.setClearingLog( true );
+    config.setSafeModeEnabled( false );
+    config.setGatheringMetrics( false );
+    config.setLogLevel( LogLevel.BASIC );
+
+    // Initialize empty collections
+    config.setVariables( new HashMap<>() );
+    config.setParams( new HashMap<>() );
+
+    return config;
+  }
+
+  /**
+   * Create an execution configuration for remote execution.
+   */
+  public static TransExecutionConfiguration createRemoteExecutionConfiguration( SlaveServer slaveServer ) {
+    TransExecutionConfiguration config = createDefaultExecutionConfiguration();
+
+    config.setExecutingLocally( false );
+    config.setExecutingRemotely( true );
+    config.setRemoteServer( slaveServer );
+
+    return config;
+  }
+
+  /**
+   * Create an execution configuration for clustered execution.
+   */
+  public static TransExecutionConfiguration createClusteredExecutionConfiguration() {
```

Comment:
TransExecutionConfiguration#ccreateClusteredExecutionConfiguration only used in test code, please remove from source code and move this logic to the test code.